frontend/public/components/resource-pages.ts

```diff
@@ -34,57 +36,104 @@ import { HorizontalPodAutoscalersPage, HorizontalPodAutoscalersDetailsPage } fro
 import { SubscriptionsPage, SubscriptionDetailsPage } from './cloud-services/subscription';
 import { InstallPlansPage, InstallPlanDetailsPage } from './cloud-services/install-plan';
 import { CatalogSourceDetailsPage } from './cloud-services/catalog-source';
 import { StorageClassPage, StorageClassDetailsPage } from './storage-class';
 
-export const resourceListPages = new Map<string, React.ComponentType<any>>()
-  .set('Clusters', ClustersPage)
-  .set('ConfigMaps', ConfigMapsPage)
-  .set('DaemonSets', DaemonSetsPage)
-  .set('DeploymentConfigs', DeploymentConfigsPage)
-  .set('Deployments', DeploymentsPage)
-  .set('BuildConfigs', BuildConfigsPage)
-  .set('Builds', BuildsPage)
-  .set('ImageStreams', ImageStreamsPage)
-  .set('Jobs', JobsPage)
-  .set('CronJobs', CronJobsPage)
-  .set('Projects', ProjectsPage)
-  .set('Namespaces', NamespacesPage)
-  .set('NetworkPolicies', NetworkPoliciesPage)
-  .set('Nodes', NodesPage)
-  .set('Pods', PodsPage)
-  .set('ReplicaSets', ReplicaSetsPage)
-  .set('ReplicationControllers', ReplicationControllersPage)
-  .set('Secrets', SecretsPage)
-  .set('ServiceAccounts', ServiceAccountsPage)
-  .set('Services', ServicesPage)
-  .set('Ingresses', IngressesPage)
-  .set('Routes', RoutesPage)
-  .set('Roles', RolesPage)
-  .set('RoleBindings', RoleBindingsPage)
-  .set('EtcdClusters', EtcdClustersPage)
-  .set('Prometheuses', PrometheusInstancesPage)
-  .set('ServiceMonitors', ServiceMonitorsPage)
-  .set('Alertmanagers', AlertManagersPage)
-  .set('PodVulns', PodVulnsPage)
-  .set('StatefulSets', StatefulSetsPage)
-  .set('ResourceQuotas', ResourceQuotasPage)
-  .set('HorizontalPodAutoscalers', HorizontalPodAutoscalersPage)
-  .set('PersistentVolumes', PersistentVolumesPage)
-  .set('PersistentVolumeClaims', PersistentVolumeClaimsPage)
-  .set('Reports', ReportsPage)
-  .set('ReportGenerationQuerys', ReportGenerationQueriesPage)
+import { referenceForModel, GroupVersionKind } from '../module/k8s';
+import {
+  ClusterModel,
+  ConfigMapModel,
+  DaemonSetModel,
+  DeploymentModel,
+  DeploymentConfigModel,
+  BuildConfigModel,
+  BuildModel,
+  ImageStreamModel,
+  JobModel,
+  CronJobModel,
+  ProjectModel,
+  NamespaceModel,
+  NetworkPolicyModel,
+  NodeModel,
+  PodModel,
+  ReplicaSetModel,
+  ReplicationControllerModel,
+  SecretModel,
+  ServiceAccountModel,
+  ServiceModel,
+  IngressModel,
+  RouteModel,
+  RoleModel,
+  RoleBindingModel,
+  EtcdClusterModel,
+  PrometheusModel,
+  ServiceMonitorModel,
+  AlertmanagerModel,
+  PodVulnModel,
+  StatefulSetModel,
+  ResourceQuotaModel,
+  HorizontalPodAutoscalerModel,
+  PersistentVolumeModel,
+  PersistentVolumeClaimModel,
+  StorageClassModel,
+  CustomResourceDefinitionModel,
+  ClusterServiceVersionModel,
+  SubscriptionModel,
+  InstallPlanModel,
+  ImageStreamTagModel,
+  ClusterRoleModel,
+  ContainerModel,
+  CatalogSourceModel,
+} from '../models';
```

Comment:
Should we consider `import * as models from '../models';`?

Reply:
I'd rather be explicit about the models we import (not all of them have resource list/detail views).